x-pack/plugins/index_management/test/fixtures/template.ts

```diff
@@ -5,39 +5,47 @@
  */
 
 import { getRandomString, getRandomNumber } from '../../../../test_utils';
-import { TemplateDeserialized } from '../../common';
+import { TemplateDeserialized, TemplateType, TemplateListItem } from '../../common';
+
+const objHasProperties = (obj?: Record<string, any>): boolean => {
+  return obj === undefined || Object.keys(obj).length === 0 ? false : true;
```

Comment:
Not sure this function name and the logic match up for me. I would expect `objHasProperties` to return `false` for when the value passed in is `undefined`.

`Object.keys(obj).length === 0 ? false : true` can be written as `Object.keys(obj).length !== 0`

Reply:
> I would expect objHasProperties to return false for when the value passed in is undefined.

🤔 hmmm isn't it what this code does?

Reply:
Maybe the code could be made a bit more verbose to make its behavior more obvious?
```ts
const objHasProperties = (obj?: Record<string, any>): boolean => {
  if (obj === undefined) {
    return false;
  }

  if (Object.keys(obj).length === 0) {
    return false;
  }

  return true;
}
```

Reply:
@sebelga I see that I misread the precedence of the evaluation.

I thought `obj === undefined` would short-circuit `Object.keys(obj).length === 0...` check. That is also why I recommended changing it to `Object.keys(obj).length !== 0`. If prettier allows, adding `()` around the predicate could also help.

Reply:
Actually to clarify; I am happy with `return obj !== undefined && Object.keys(obj).length !== 0`. Or CJ's suggestion :)

Reply:
What is wrong with the "or" operator? 😄 

```
js
const objHasProperties = (obj?: Record<string, any>): boolean => {
  if (obj === undefined || Object.keys(obj).length === 0) {
    return false;
  }

  return true;
}
```

And then, I decided to use a ternary to remove the if clause.

Reply:
> I does short-circuit so it does not execute Object.keys(...) on an undefined value

I see my comment was not very clear,  I meant short circuit the whole -> `Object.keys(obj).length === 0 ? false : true` part of the line (which is wrong). This is a case where the JS grammar seems ambiguous to me.

I'll leave it up to you!